### PR TITLE
Sp-714 reversals

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: Library/SPIClient.m [hunk 1]
```diff
@@ -1399,6 +1441,37 @@ - (void)handleRefundResponse:(SPIMessage *)m {
     [self handleTxResponse:m type:SPITransactionTypeRefund checkPosRefId:YES];
 }
 
+- (void)handleReversalResponse:(SPIMessage *)m {
+    
+    NSLog(@"handleReversalResponse");
+    NSLog(@"hansdleReversalResp txLock entering");
```
Comment (mattgresham) [inline Library/SPIClient.m]: Small typo here
```suggestion
    NSLog(@"handleReversalResp txLock entering");
```

File: Library/SPIClient.m [hunk 2]
```diff
@@ -1399,6 +1441,37 @@ - (void)handleRefundResponse:(SPIMessage *)m {
...
+    @synchronized(self.txLock) {
+        NSLog(@"handleReversalResp txLock entered");
+        SPITransactionFlowState *txState = self.state.txFlowState;
+        NSDictionary *dict = m.data;
+        NSString *incomingPosRefId = [dict objectForKey:@"pos_ref_id"];
+        
+        SPIReversalResponse *revResp = [[SPIReversalResponse alloc] initWithMessage:m];
+        
+        if (self.state.flow != SPIFlowTransaction || txState.isFinished || txState.posRefId == incomingPosRefId) {
+            SPILog(@"Received Reversal response but I was not waiting for this one.");
```
Comment (mattgresham) [inline Library/SPIClient.m]: Can you please add the `incomingPosRefId` to the end of this message too? Something like "`Incoming Pos Ref ID: {incomingPosRefId}`"

**Review by mattgresham — COMMENTED** [review]
Looks good to me. Just a couple of small things

File: Library/SPIClient.m (hunk 2 above)
Comment (ghost) [reply Library/SPIClient.m]: Got it

**Review by mattgresham — APPROVED** [review]

**Pull request merged**